x/observer/keeper/ballot.go

```diff
@@ -58,10 +71,25 @@ func (k Keeper) GetAllBallots(ctx sdk.Context) (voters []*types.Ballot) {
 
 // AddBallotToList adds a ballot to the list of ballots for a given height.
 func (k Keeper) AddBallotToList(ctx sdk.Context, ballot types.Ballot) {
-	list, found := k.GetBallotList(ctx, ballot.BallotCreationHeight)
+	list, found := k.GetBallotListForHeight(ctx, ballot.BallotCreationHeight)
 	if !found {
 		list = types.BallotListForHeight{Height: ballot.BallotCreationHeight, BallotsIndexList: []string{}}
 	}
 	list.BallotsIndexList = append(list.BallotsIndexList, ballot.BallotIdentifier)
 	k.SetBallotList(ctx, &list)
 }
+
```

Comment:
We should add a function allowing to delete all matured ballots ever and run it in the migration. We should also check efficiency in the mainnet upgrade beforehand, might be long with all the ballots

Reply:
I don't know if deleting the older ballots would help with the archive node's growth; non-archives would get pruned at some point. 
Happy to add that in and test if you think we still need it

Reply:
This will allow to have a cleaner state in general, we can then query all the existing ballots to get the active ones.

The state of the store would make less sense otherwise, you would have active ballots but still matured ballots for older blocks?

Reply:
Hmm , I guess we could have a query for all ballots if we delete the older ones, 
Ill look into testing the upgrade handler then

Reply:
Tested with mainnet data , the delete time is less than 1s
The data is is ~700MB 

We are waiting for the test snapshot to be available before resolving this comment. The testnet data is too large to be able to parse easily , I am working with Julian to extract just the observer data , instead of the whole export